src/convertToObject.js

```diff
@@ -5,8 +5,21 @@
  *
  * @return {object}
  */
+
 function convertToObject(sourceString) {
-  // write your code here
+  const obj = sourceString
+    .split(';')
+    .filter((line) => line.trim() !== '')
+    .reduce((object, el) => {
+      let [key, value] = el.split(':').map((item) => item.trim());
+
+      key = key.split('').join('').trim();
```

Comment:
[CODE KNOWLEDGE] - The operation `key.split('').join('')` is redundant since it effectively returns the original string. You can remove this line as it does not modify the `key` in any way.